catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/view/Segment.view.js

```diff
@@ -0,0 +1,183 @@
+                "click .add-segment": 'addSegment'
+            },
+            initialize: function (options) {
+                this.editableSegment = options.editableSegment;
+                this.listenTo(wreqr.vent, 'fieldChanged:' + this.model.get('segmentId'), this.updateTitle);
+                this.addRegions({
+                    associations: '#associations-' + this.model.get('simpleId'),
+                    customizable: '#customizable-' + this.model.get('simpleId')
+                });
+            },
+            onRender: function () {
+                var knownFields = [];
+                var allFields = this.model.get('fields').models;
+                _.each(allFields, function (field) {
+                    if (!field.get('custom')) {
+                        knownFields.push(field);
+                    }
+                });
+                this.formFields.show(new Field.FieldCollectionView({collection: new Backbone.Collection(knownFields)}));
+                this.formSegments.show(new Segment.SegmentCollectionView({
+                    model: this.model,
+                    collection: this.model.get('segments'),
+                    showHeader: knownFields.length > 0 ? false : true
+                }));
+                if (FieldDescriptors.isCustomizableSegment(this.model.get('segmentType'))) {
+                    this.customizable.show(new Customizable.CustomizableCollectionView({
+                        model: this.model
+                    }));
+                    if (this.model.get('fields').models.length > 0) {
+                        var associationModel = this.model.get('associationModel');
+                        this.associations.show(new Association.AssociationCollectionView({
+                            parentId: this.model.get("segmentId"),
+                            model: associationModel,
+                            collection: new Backbone.Collection(associationModel.getAssociationsForId(this.model.get('segmentId')))
+                        }));
+                    }
+                }
+                this.setupPopOvers();
+            },
+            removeSegment: function () {
+                wreqr.vent.trigger('removeSegment:' + this.model.get('parentId'), this.model.get('segmentId'));
+            },
+            addSegment: function () {
+                this.model.addSegment();
+                this.render();
+            },
+            updateTitle: function () {
+                if (!this.model.get('multiValued') && this.model.constructTitle) {
+                    var title = this.$('.segment-title-'+this.model.get('simpleId'));
+                    title.html(this.model.constructTitle());
+                }
+            },
+            serializeData: function () {
+                var data = {};
+
+                if (this.model) {
+                    data = this.model.toJSON();
+                }
+                data.editableSegment = this.editableSegment;
```

Comment:
This isn't the correct way to do this. Why not just have editableSegment in the model have the data that you want?